### PR TITLE
Add hard fork code and new release version

### PR DESCRIPTION
**Pull request by awfm9**

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -189,11 +189,8 @@ func (n *Node) initNetworking() error {
 
 	tlsConfig := network.TLSConfig(n.Config.StakingTLSCert)
 
-	// Initialize validator manager and primary network's validator set
-	n.validators = validators.NewDefaultSet(n.Config.NetworkID)
-	if n.validators.Len() == 0 {
-		return fmt.Errorf("no default validator list (network: %d)", n.Config.NetworkID)
-	}
+	// Initialize default validator set for network
```
Comment (emailtovamos) [inline node/node.go:192]: Should the comment include `empty validator set` as `NewSet()` returns an empty set?

File: version/constants.go [hunk 2]
```diff
@@ -18,20 +18,12 @@ import (
...
-
-	// Legacy versioning constants.
-	Legacy                      = NewDefaultVersion(1, 7, 5)
-	LegacyApp                   = NewDefaultApplication(constants.LegacyPlatformName, Legacy.Major(), Legacy.Minor(), Legacy.Patch())
-	LegacyCompatibleVersion     = NewDefaultApplication(constants.LegacyPlatformName, 1, 7, 0)
-	PrevLegacyCompatibleVersion = NewDefaultApplication(constants.LegacyPlatformName, 1, 5, 0)
-	LegacyUnmaskedVersion       = NewDefaultApplication(constants.LegacyPlatformName, 1, 1, 0)
-	PrevLegacyUnmaskedVersion   = NewDefaultApplication(constants.LegacyPlatformName, 1, 0, 0)
+	MinimumCompatibleVersion     = NewDefaultApplication(constants.PlatformName, 0, 5, 1)
+	PrevMinimumCompatibleVersion = NewDefaultApplication(constants.PlatformName, 0, 5, 1)
```
Comment (emailtovamos) [inline version/constants.go:24]: Why `PrevMinimumCompatibleVersion` and `MinimumCompatibleVersion` are the same?

File: network/network.go [hunk 3]
```diff
@@ -557,8 +525,7 @@ func (n *network) Dispatch() error {
 	go n.inboundConnUpgradeThrottler.Dispatch()
 	defer n.inboundConnUpgradeThrottler.Stop()
 	go func() {
-		compatibility := n.compatibility()
```
Comment (emailtovamos) [inline network/network.go:560]: I didn't quite understand the places where `compatibility()` is replaced by `versionCompatibility`. Could you elaborate on why it was done?

File: version/constants.go [hunk 4]
```diff
@@ -132,15 +137,3 @@ func GetCompatibility(networkID uint32) Compatibility {
 		PrevMinimumUnmaskedVersion,
 	)
 }
-
-func GetLegacyCompatibility(networkID uint32) Compatibility {
```
Comment (emailtovamos) [inline version/constants.go:136]: Will we need something like this when we go on to upgrade in future (e.g. `FlareHardFork2`)?

File: version/constants.go [hunk 5]
```diff
@@ -121,6 +119,13 @@ func GetApricotPhase5Time(networkID uint32) time.Time {
 	return ApricotPhase5DefaultTime
 }
 
+func GetFlareHardFork1Time(networkID uint32) time.Time {
+	if upgradeTime, exists := ApricotPhase5Times[networkID]; exists {
```
Comment (emailtovamos) [inline version/constants.go:123]:
```suggestion
	if upgradeTime, exists := FlareHardFork1Times[networkID]; exists {
```

File: network/peer.go [hunk 6]
```diff
@@ -601,12 +591,8 @@ func (p *peer) handleGetVersion(_ message.InboundMessage) {
 
 // assumes the [stateLock] is not held
 func (p *peer) handleVersion(msg message.InboundMessage) {
-
-	// We always allow the first version message through; if we have already processed
-	// the first version message, we let the message through only if the version
-	// reset was requested, but the second version message not yet processed.
 	switch {
-	case p.gotVersion.GetValue() && (!p.resetVersion.GetValue() || p.regotVersion.GetValue()):
```
Comment (Ullaakut) [inline network/peer.go:609]: Why this change? It seems like they had a reason to do this specific check in this condition but based on the context I have at the time of writing this comment, I don't see why the changes of the PR would make this obsolete

File: network/network.go (hunk 3 above)
Comment (awfm9) [reply network/network.go:560]: This was a change I originally did myself, to have support for old nodes, which were still sending legacy Avalanche version information, and new nodes, which were sending the correct Flare version information. Now that all nodes are on the new code, we can drop support for the legacy version.

`compatibility()` picked one compatibility version based on what hard fork we were at.

`compatibilities()` picked one or two compatibility versions based on what hard fork we were at, and had accompanying functions to mirror the compatibility functions for more than one compatibility version.

File: network/peer.go (hunk 6 above)
Comment (awfm9) [reply network/peer.go:609]: I implemented this myself. This code allowed nodes to request a version reset once, which was needed because of a weird unnecessary check in the Avalanche networking code. On every `pong` we receive as an answer to a `ping`, the node checks if the version is compatible, and will disconnect if not. This check is strange, because in practice, Avalanche nodes should _never_ change their version unless they restart/disconnect.

However, this broke our code, where nodes started sending new version information after one hard fork, and then started rejecting the old version information after the next hard fork, leading to a disconnect for everyone on hard fork. To minimize disruption to operators, I implemented this fix before we deployed on Songbird, and no one got disconnected.

Now, as everyone is on a new stable version, that will not change except with code upgrades, it's no longer needed.

File: version/constants.go (hunk 2 above)
Comment (awfm9) [reply version/constants.go:24]: We are starting a new lineage of versioning. There was no version before `v0.5.1` that was available to the public. `v0.5.0` could be part of the peer-to-peer network as the code stands, so I left the unmasked version at that.

File: version/constants.go (hunk 5 above)
Comment (awfm9) [reply version/constants.go:123]: Good catch!

File: version/constants.go (hunk 4 above)
Comment (awfm9) [reply version/constants.go:136]: No, because versions that nodes send will now always be aligned with the code version, so they have to upgrade the binary and restart the node anyway.
